db/hints/sync_point.cc

```diff
@@ -69,13 +72,30 @@ static std::vector<sync_point::shard_rps> decode_one_type_v1(uint16_t shard_coun
     return ret;
 }
 
+static uint64_t calculate_checksum(const sstring_view s) {
+    xx_hasher h;
+    h.update(s.data(), s.size());
+    return h.finalize_uint64();
+}
+
 sync_point sync_point::decode(sstring_view s) {
     bytes raw = base64_decode(s);
     sstring_view raw_s(reinterpret_cast<char*>(raw.data()), raw.size());
     seastar::simple_memory_input_stream in{raw_s.data(), raw_s.size()};
 
     uint8_t version = ser::serializer<uint8_t>::read(in);
-    if (version != 1) {
+    if (version == 2) {
+        if (raw_s.size() < version_size + checksum_size) {
+            throw std::runtime_error("Could not decode the sync point encoded in the V2 format - not a valid hex string");
```

Comment:
It looks like you copied the old error message here which used to be thrown when `s` is empty, but I think it was slightly wrong and here we have a chance to improve it. How about:

```
Could not decode the sync point encoded in the V2 format - serialized blob is too short
```